### DISCUSSION TITLE
Issue writing to raster after reading from a raster

### DISCUSSION THREAD

**Discussion by kylebradbury** [corteva/rioxarray#712] [Q&A] [answered]
Hi All - I'm not sure if this is user error or a bug, so I figured I'd start here. I'm trying to load a GeoTIFF, do some processing to the contents, and then write back to a GeoTIFF, but I'm consistently getting an error. Below is a simple example of reproducing the issue:
```
import rioxarray
file = "./example.tif" # A valid GeoTIFF file
xds = rioxarray.open_rasterio(file, engine='rasterio')
xds.rio.to_raster('test.tif')
```
And here's the content of the file, summarized:
<img width="707" alt="Screenshot 2023-11-17 at 1 26 41 PM" src="https://github.com/corteva/rioxarray/assets/4689093/07552dc9-b2f8-45bb-ba89-97713f56cf27">

Thanks in advance for your help.

**Comment by snowman2** [comment]
Possible to share the file?

Also, what is the error you are getting?

**Reply by kylebradbury** [reply]
Thanks for responding!

The error I'm receiving is:
`AttributeError: 'DataArray' object has no attribute '_data'`

I've [shared the file here](https://www.dropbox.com/scl/fi/n87348s49fzd2px9cjaaw/example.tif?rlkey=y763twokk8u0ianj1eaw6mlrc&dl=0) (it's about 200 Mb)

**Comment by RichardScottOZ** [comment]
Worked for me @kylebradbury 

![image](https://github.com/corteva/rioxarray/assets/72196131/9821f31d-c792-4d75-90db-4e226e2301ce)

**Comment by RichardScottOZ** [comment]
what version of rioxarray - note didn't use the engine form of opening - not sure if that could be different?

**Comment by kylebradbury** [comment, marked as answer]
You're right - this was a version issue. I had 0.13.1. I had been using mamba to update the package and it wasn't recognizing that an update was available. I had to force the version change and once I upgraded to version 0.15.1, this worked fine. Thank you for your help!
